File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.java
class Solution {
    public boolean halvesAreAlike(String s) {
          Set<Character> vowels = Set.of('a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U');
        int vowelsCount = 0, midIndex = s.length() / 2;

        for (int i = 0; i < midIndex; i++) {
            char charA = s.charAt(i);
            char charB = s.charAt(midIndex + i);
            if (vowels.contains(charA)) vowelsCount++;
            if (vowels.contains(charB)) vowelsCount--;
        }
        return vowelsCount == 0;
    
    }
}

File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.py
class Solution:
    def halvesAreAlike(self, s: str) -> bool:
        vowels=['a','e','i','o','u','A','E','I','O','U']
        count1,count2=0,0
        a=len(s)//2
        print(a)
        for i in range(a):
            if s[i] in vowels:
                count1+=1
        for i in range(a,len(s)):
            if s[i] in vowels:
                count2+=1
        return count1==count2